preview_generator/preview/builder/image__pillow.py

```diff
@@ -59,23 +67,43 @@ def image_to_jpeg_pillow(
                 dims_in=ImgDims(width=image.size[0], height=image.size[1]),
                 dims_out=preview_dims
             )
+            output = BytesIO()
+            image = image.resize((resize_dim.width, resize_dim.height), resample=JPEG_RESAMPLE)
+            if image.mode not in TRANSPARENCY_MODES:
+                # INFO - G.M - 2019-06-18 - Try directly saving new image, this may failed due to
+                # image mode issue.
+                try:
+                    image.save(output, 'jpeg', optimize=JPEG_OPTIMIZE, quality=JPEG_QUALITY, progressive=JPEG_PROGRESSIVE)
+                except OSError as exc:
```

Comment:
I do not understand the OSError and the new save being "only" to convert image mode (so what is the relationship with OSError?)

Reply:
When image mode is not correct, save doesn't work properly. As Pillow code doesn't allow to know clearly which mode work and which mode doesn't we should try to save as original mode and if it fail save as RGB mode which is known to work. Exception is OsError in this case which is not really nice but it doesn't have specific exception.